packages/back-end/src/models/ExperimentSnapshotModel.ts

```diff
@@ -103,7 +110,107 @@ const ExperimentSnapshotModel = mongoose.model<ExperimentSnapshotDocument>(
 
 const toInterface = (
   doc: ExperimentSnapshotDocument
-): ExperimentSnapshotInterface => omit(doc.toJSON(), ["__v", "_id"]);
+): ExperimentSnapshotInterface =>
+  migrateSnapshot(omit(doc.toJSON(), ["__v", "_id"]));
+
+export function migrateSnapshot(
+  orig: LegacyExperimentSnapshotInterface
+): ExperimentSnapshotInterface {
+  const {
+    activationMetric,
+    statsEngine,
+    // eslint-disable-next-line
+    hasRawQueries,
+    // eslint-disable-next-line
+    hasCorrectedStats,
+    // eslint-disable-next-line
+    query,
+    results,
+    regressionAdjustmentEnabled,
+    metricRegressionAdjustmentStatuses,
+    sequentialTestingEnabled,
+    sequentialTestingTuningParameter,
+    queryFilter,
+    segment,
+    skipPartialData,
+    manual,
+    ...snapshot
+  } = orig;
+
+  // Convert old results to new array of analyses
+  if (!snapshot.analyses) {
+    if (results) {
+      const regressionAdjusted =
+        regressionAdjustmentEnabled &&
+        metricRegressionAdjustmentStatuses?.some(
+          (s) => s.regressionAdjustmentEnabled
+        )
+          ? true
+          : false;
+
+      snapshot.analyses = [
+        {
+          dateCreated: snapshot.dateCreated,
+          status: snapshot.error ? "error" : "success",
+          error: snapshot.error,
+          settings: {
+            statsEngine: statsEngine || "bayesian",
+            dimensions: snapshot.dimension ? [snapshot.dimension] : [],
+            pValueCorrection: null,
+            regressionAdjusted,
+            sequentialTesting: !!sequentialTestingEnabled,
+            sequentialTestingTuningParameter:
+              sequentialTestingTuningParameter ||
+              DEFAULT_SEQUENTIAL_TESTING_TUNING_PARAMETER,
+          },
+          results,
+        },
+      ];
+    } else {
+      snapshot.analyses = [];
+    }
+  }
+
+  // Figure out status from old fields
+  if (!snapshot.status) {
+    snapshot.status = snapshot.error
+      ? "error"
+      : snapshot.analyses.length > 0
+      ? "success"
+      : "running";
+  }
+
+  // Migrate settings
+  // We weren't tracking all of these before, so just pick good defaults
+  if (!snapshot.settings) {
+    snapshot.settings = {
+      manual: !!manual,
+      dimensions: snapshot.dimension
+        ? [
+            {
+              id: snapshot.dimension,
+            },
+          ]
+        : [],
+      // TODO: figure out metrics from results?
+      goalMetrics: [],
```

Comment:
Just like with `phase` below, we would have to get the keys in `snapshot.results[0].variations[0].metrics` and then go look up the associated metrics, unless we're just storing the keys. I think since the phase and metrics could have changed since it last ran, we probably should just get the IDs in the JIT?

Not sure, just thinking out loud.